SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -687,29 +692,210 @@ def plot_3d_pseudosection(
             elif scale == "linear":
                 cbar = plt.colorbar(
-                    data_plot, format="%.2e", norm=norm, cax=cax, **cbar_opts,
+                    data_plot, format="%.2e", cax=cax, **cbar_opts,
                 )
 
-        ticks = np.linspace(norm.vmin, norm.vmax, 5)
+        ticks = np.linspace(norm.vmin, norm.vmax, 7)
 
         cbar.set_ticks(ticks)
-        cbar.set_label(units, labelpad=12)
+        cbar.set_label(units, labelpad=10)
         cbar.ax.tick_params()
 
     return ax
 
+if has_plotly:
+    def plot_3d_pseudosection(
+        survey,
+        dvec,
+        marker_size=4,
+        vlim=None,
+        scale="linear",
+        units="",
+        plane_points=None,
+        plane_distance=10.0,
+        cbar_opts=None,
+        marker_opts=None,
+        layout_opts=None
+    ):
+        """
+        Plot 3D DC/IP data in pseudo-section as a scatter plot.
+
+        This utility allows the user to produce a scatter plot of 3D DC/IP data at
+        all pseudo-locations. If a plane is specified, the user may create a scatter
+        plot using points near that plane.
+
+        Input:
+        survey : SimPEG.electromagnetics.static.survey.Survey
+            A DC or IP survey object
+        dvec : numpy.ndarray
+            A data vector containing volts, integrated chargeabilities, apparent
+            resistivities or apparent chargeabilities.
+        marker_size : int
+            Sets the marker size for the points on the scatter plot
+        vlim : list
+            list containing the minimum and maximum value for the color range,
+            i.e. [vmin, vmax]
+        scale: str
+            Plot on linear or log base 10 scale {'linear','log'}
+        units : str
+            A sting in d3 formatting the specified the units of *dvec*
+        plane_points : list of numpy.ndarray
+            A list of length 3 which contains the three xyz locations required to
+            define a plane; i.e. [xyz1, xyz2, xyz3]. This functionality is used to
+            plot only data that lie near this plane. A list of [xyz1, xyz2, xyz3]
+            can be entered for multiple planes.
+        plane_distance : float or list of float
+            Distance tolerance for plotting data that are near the plane(s) defined by
+            **plane_points**. A list is used if the *plane_distance* is different
+            for each plane.
+        cbar_opts: dict
+            Dictionary containing colorbar properties formatted according to plotly.graph_objects.scatter3d.cbar
+        marker_opts : dict
+            Dictionary containing marker properties formatted according to plotly.graph_objects.scatter3d
+        layout_opts : dict
+            Dictionary defining figure layout properties, formatted according to plotly.Layout
+        
+
+        Output:
+        fig: 
+            A plotly figure
+
+        """
+
+        locations = pseudo_locations(survey)
+
+        # Scaling
+        if scale == "log":
```

Comment:
I tried looking, but it doesn't look like there's any way to supply a "norm" like argument to poorly for plotting on a log scale like there is in numpy. I would also like to see a better format than plotting as "10^###" as it is very confusing. I imagine there must be some way with plotly to directly control the tick labels, and replace 10^### with an actual interpretable number.

Reply:
See above comment